urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlib/property/Vexillary.java

```diff
@@ -0,0 +1,303 @@
+package permlib.property;
+
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Collection;
+import permlib.PermUtilities;
+import permlib.Permutation;
+
+/**
+ *
+ * @author MichaelAlbert
+ */
+public class Vexillary implements HereditaryProperty {
+
+    static HereditaryProperty A2143 = AvoidanceTest.getTest("2143");
+    private int bound;
+    ArrayDeque<Permutation> q = new ArrayDeque<Permutation>();
+    ArrayDeque<Permutation> children = new ArrayDeque<Permutation>();
+    ArrayDeque<int[]> arrayChildren = new ArrayDeque<int[]>();
+    ArrayDeque<int[]> aq = new ArrayDeque<int[]>();
+
+    public Vexillary(int bound) {
+        this.bound = bound;
+    }
+
+    public Collection<Permutation> getBasis() {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public Collection<Permutation> getBasisTo(int n) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public boolean isSatisfiedBy(Permutation p) {
+        return vexBound(p, bound);
+    }
+
+    @Override
+    public boolean isSatisfiedBy(int[] values) {
+        return vexBound(values, bound);
+    }
+
+    public ArrayDeque<Permutation> children(Permutation p) {
+        children.clear();
+        if (A2143.isSatisfiedBy(p.elements)) {
+            return children;
+        }
+//        int pInv = p.inversions();
+//        System.out.println("p = " + p);
+        RS rs = rs(p);
+//        System.out.println("r = " + rs.r + " s = " + rs.s);
+//        for (int j = 0; j < rs.r; j++) {
+//            if (j != rs.r) {
+//                Permutation q = p.clone();
+//                int v = q.elements[j];
+//                q.elements[j] = q.elements[rs.s];
+//                q.elements[rs.s] = q.elements[rs.r];
+//                q.elements[rs.r] = v;
+//                // System.out.println(q + " " + q.inversions());
+//                if (q.inversions() == pInv) {
+//                    result.add(q);
+//                }
+//            }
+//        }
+        int deltaS = 0;
+        for (int j = rs.r - 1; j >= 0; j--) {
+            int[] elements = p.elements;
+            if (elements[j] < elements[rs.s]) {
+                deltaS++;
+            } else {
+                deltaS--;
+            }
+            int deltaJ = 0;
+            for (int t = j + 1; t < rs.r; t++) {
+                if (elements[t] > elements[j]) {
+                    deltaJ++;
+                } else {
+                    deltaJ--;
+                }
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 76:82, 129:135.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.